### PR TITLE
Exposed routes as JSON in RouteCompletionHandler

### PR DESCRIPTION
**Pull request by vincethecoder**
This exposes a third argument in the `RouteCompletionHandler` tuple; `JSONRoutes`. This represents the returned routes in an array of `[String: Any]` dictionaries. 

Addresses https://github.com/mapbox/mapbox-navigation-ios/pull/1496

### CODE REVIEW

File: MapboxDirections/MBDirections.swift [hunk 1]
```diff
@@ -103,7 +103,7 @@ open class Directions: NSObject {
         If the request was canceled or there was an error obtaining the routes, this parameter is `nil`. This is not to be confused with the situation in which no results were found, in which case the array is present but empty.
      - parameter error: The error that occurred, or `nil` if the placemarks were obtained successfully.
      */
-    public typealias RouteCompletionHandler = (_ waypoints: [Waypoint]?, _ routes: [Route]?, _ error: NSError?) -> Void
+    public typealias RouteCompletionHandler = (_ waypoints: [Waypoint]?, _ routes: [Route]?, _ JSONRoutes: [[String: Any]]?, _ error: NSError?) -> Void
```
Comment (bsudekum) [inline MapboxDirections/MBDirections.swift]: Is there any way we could make this a non breaking change? What if JSONRoutes was nil by default?

Comment (1ec5) [reply MapboxDirections/MBDirections.swift]: That would still be a breaking change. There’s no way to add an argument to this closure without breaking backwards compatibility.

File: MapboxDirections/MBDirections.swift [hunk 2]
```diff
@@ -103,7 +103,7 @@ open class Directions: NSObject {
         If the request was canceled or there was an error obtaining the routes, this parameter is `nil`. This is not to be confused with the situation in which no results were found, in which case the array is present but empty.
      - parameter error: The error that occurred, or `nil` if the placemarks were obtained successfully.
```
Comment (vincethecoder) [inline MapboxDirections/MBDirections.swift]: TODO: 

- [ ] Add a description for the new field `_ JSONRoutes: [[String: Any]]?` within the `RouteCompletionHandler`

**Comment by frederoni** [comment]
This was superseded by https://github.com/mapbox/MapboxDirections.swift/pull/297

**Pull request closed without merging**